src/Viewer.tsx

```diff
@@ -317,14 +323,78 @@ function Viewer(): ReactElement {
     [featureThresholds, config.keepRangeBetweenDatasets]
   );
 
+  // DATASET LOADING ///////////////////////////////////////////////////////
+  /**
+   * Replaces the current dataset with another loaded dataset. Handles cleanup and state changes.
+   * @param newDataset the new Dataset to replace the existing with. If null, does nothing.
+   * @param newDatasetKey the key of the new dataset in the Collection.
+   * @returns a Promise<void> that resolves when the loading is complete.
+   */
+  const replaceDataset = useCallback(
```

Comment:
Moved from below without changes because it was previously being called before declaration which can introduce bugs.